superset/models/helpers.py

```diff
@@ -212,7 +212,7 @@ def export_to_dict(self, recursive=True, include_parent_ref=False,
                             include_defaults=include_defaults,
                         ) for child in getattr(self, c)
                     ],
-                    key=lambda k: sorted(k.items()))
+                    key=lambda k: sorted(str(k.items())))
```

Comment:
Wow this method is pretty heavy/opaque.  I'm confused as to why we sort. I'm guessing it's to have the nested object be somewhat deterministic (though not sure why it matters). Now why sorting based on the `items` makes no sense to me. Why not `lambda k: k.get('id')` or something like that? If `id` isn't exported, maybe it could be added.

But does the ordering matter here?

Reply:
Comment points to @fabianmenges, do you have more context?

Reply:
It seems like we want the order to be deterministic - a pretty reasonable thing to want in a data export. Sorting by id would be better.

Reply:
Sorting by the list of items makes no sense though. If the list is 
```python
[
    {'id': 2, 'name': 'test2'},
    {'id': 1, 'name': 'test2'},
]
```
it would sort based on 
```
[
    [2, 'test2'],
    [1, 'test1'],
]
```
which is totally weird, logic is flawed here

Reply:
Yeah, on a first pass, I couldn't figure out what it was trying to do and did something that wouldn't change the behavior. Probably should have used blame here. I'll take another look at resolving this.